tests/test_lexer.py

```diff
@@ -88,6 +88,9 @@
         ('foo "true"', ['STRING', 'STRING'], ['foo', 'true']),
         ('foo:bar:', ['STRING', 'COLON', 'STRING', 'COLON'], ['foo', ':', 'bar', ':']),
         ('"foo:bar":', ['STRING', 'COLON'], ['foo:bar', ':']),
+        ('true false', ['BOOLEAN', 'BOOLEAN'], [True, False]),
+        ('true-false', ['BOOLEAN', 'STRING'], [True, '-false']),
```

Comment:
Shouldn't this be just a unique string?

i.e., from the commit msg:

> Previously, pyarn would not tokenize booleans at the beginning of a
> longer string. Example: 'true-case-path' (should be BOOLEAN STRING,
> pyarn thought it was just one STRING).

Why can't `true-case-path` be a string?

Reply:
Hey Athos :-)

Because `yarn` itself will tokenize it as BOOLEAN STRING

You can check with this yarn.lock:
```
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1

node-sass@^4.14.1:
  version "4.14.1"
  dependencies:
    true-case-path "^1.0.2"
```
(you'll need a package.json to go with it)
```json
{
  "name": "foo",
  "version": "1.0.0",
  "dependencies": {
    "node-sass": "^4.14.1"
  }
}
```

Reply:
Aha!

```
warning package.json: No license field
error An unexpected error occurred: "Unknown token: { line: 599, col: 5, type: 'BOOLEAN', value: true } 599:5 in $SOMEPATH/yarn.lock".
info If you think this is a bug, please open a bug report with the information provided in "$SOMEPATH/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

It seems that this is intrinsic to the (naive) way yarn v1 used to parse those strings to extract booleans out of it.

Wouldn't just keeping the change in `pyarn/lockfile.py` from this PR be enough to fix this case though? The lexer/parsers here are definitely not implementing the exact same grammar as yarn v1 and this would avoid the amiguity described in the `pyarn/lexer.py` change's comment.

In either case, this LGTM (not that my review counts for anything anyway :)

Reply:
@athos-ribeiro Adam shared this with me https://github.com/yarnpkg/yarn/blob/master/src/lockfile/parse.js#L125
You are right about 
> It seems that this is intrinsic to the (naive) way yarn v1 used to parse those strings to extract booleans out of it.

Reply:
> Wouldn't just keeping the change in `pyarn/lockfile.py` from this PR be enough to fix this case though? The lexer/parsers here are definitely not implementing the exact same grammar as yarn v1 and this would avoid the ambiguity described in the `pyarn/lexer.py` change's comment.

@athos-ribeiro It would be enough, but then I'd have to justify why quoting follows different rules than parsing. I would rather keep it consistent.